sdk/ahnlich-client-py/internals/builders/db.py

```diff
@@ -0,0 +1,147 @@
+        condition: query.PredicateCondition = None,
+    ):
+        nonzero = NonZeroSizeInteger(closest_n)
+        self.queries.append(
+            query.Query__GetSimN(
+                store=store_name,
+                search_input=search_input,
+                closest_n=nonzero.value,
+                algorithm=algorithm,
+                condition=condition,
+            )
+        )
+
+    def create_index(self, store_name: str, predicates: typing.Sequence[str]):
+        self.queries.append(
+            query.Query__CreateIndex(store=store_name, predicates=predicates)
+        )
+
+    def drop_index(
+        self,
+        store_name: str,
+        predicates: typing.Sequence[str],
+        error_if_not_exists: bool,
+    ):
+        self.queries.append(
+            query.Query__DropIndex(
+                store=store_name,
+                predicates=predicates,
+                error_if_not_exists=error_if_not_exists,
+            )
+        )
+
+    def set(
+        self,
+        store_name,
+        inputs: typing.Sequence[typing.Tuple[query.Array, typing.Dict[str, str]]],
+    ):
+        self.queries.append(query.Query__Set(store=store_name, inputs=inputs))
+
+    def delete_key(self, store_name: str, keys: typing.Sequence[query.Array]):
+        self.queries.append(query.Query__DelKey(store=store_name, keys=keys))
+
+    def delete_predicate(self, store_name: str, condition: query.PredicateCondition):
+        self.queries.append(query.Query__DelPred(store=store_name, condition=condition))
+
+    def drop_store(self, store_name: str, error_if_not_exists: bool):
+        self.queries.append(
+            query.Query__DropStore(
+                store=store_name, error_if_not_exists=error_if_not_exists
+            )
+        )
+
+    def list_stores(self):
+        self.queries.append(query.Query__ListStores())
+
+    def info_server(self):
+        self.queries.append(query.Query__InfoServer())
+
+    def list_clients(self):
+        self.queries.append(query.Query__ListClients())
+
+    def ping(self):
+        self.queries.append(query.Query__Ping())
+
+    def drop(self):
+        self.queries.clear()
+
+    def to_server_query(self) -> query.ServerQuery:
+        if not self.queries:
+            raise ah_exceptions.AhnlichClientException(
+                "Must have atleast one request to be processed"
+            )
+        # not optimal, but so far, recreating the list and dropping the internal store.
+        # seems straight forward
+        queries = self.queries[:]
+
+        server_query = query.ServerQuery(queries=queries)
+        self.drop()
+        return server_query
+
+    def execute_requests(self, client: AhnlichProtocol):
+        response = client.process_request(message=self.to_server_query())
+        return response
+
+
+def create_store_key(data: typing.List[float], v: int = 1) -> query.Array:
+    np_array = np.array(data, dtype=np.float32)
+    dimensions = (st.uint64(np_array.shape[0]),)
```

Comment:
Why is dimension 0?
Shouldn't it be Array1 which should be (N, 1) where N is the length of data?

Reply:
np_array.shape returns the (NxM) of a array where N is the length of Rows and M is the length of columns.
> For a 1D array, the shape would be (n,) where n is the number of elements in your array.
> For a 2D array, the shape would be (n,m) where n is the number of rows and m is the number of columns in your array.

To your question, I'm essentially getting the first element of the shape tuple (n,m) to create a tuple of (n,)